src/main/kotlin/xyz/atrius/waystones/handler/LinkHandler.kt

```diff
@@ -23,14 +23,14 @@ class LinkHandler(
             return Ignore
         // Check if the player is able to link to this waystone
         if (!player.hasPermission("waystones.link"))
-            return Fail("This link has severed and rotted away...")
+            return Fail(localization["link-no-permission"].toString())
         // Prevent linking if relinking is disabled
         val meta = item.itemMeta as CompassMeta
         return if (!configuration.relinkableKeys() && meta.hasLodestone())
-            Fail("The destination for this key has been sealed")
+            Fail(localization["link-not-relinkable"].toString())
         // Prevent relinking if the location is the same
         else if (!player.immortal && meta.lodestone == block.location)
-            Fail("Already linked to this destination")
+            Fail(localization["link-already-linked"].toString())
```

Comment:
We seem to be calling `toString()` a lot; i'm not sure how much I like this approach. We might need to find a better way to handle this or possibly convert any message properties to `LocalizedString`.

Reply:
Now the Fail() takes a `LocalizedString` instead, no more `toString()` other than in utility classes.